packages/subcommands/src/register.ts

```diff
@@ -0,0 +1,44 @@
+import { Plugin, postInitialization, postLogin, SapphireClient } from '@sapphire/framework';
+import { SubcommandsPluginOptions, SubcommandsHandler } from './lib/utils/SubcommandsHandler';
+import { join } from 'path';
+import type { ClientOptions } from 'discord.js';
+
+export class Subber extends Plugin {
+	public static [postInitialization](this: SapphireClient, options: ClientOptions): void {
+		this.subcommandsHandler = new SubcommandsHandler(options.subcommands);
+		this.events.registerPath(join(__dirname, 'events'));
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	public static [postLogin](this: SapphireClient, _options: ClientOptions): void {
+		this.subcommandsHandler.buildSubcommandsTree(this.commands);
+	}
+}
+
+declare module '@sapphire/framework' {
+	export interface CommandOptions {
+		subcommands?: string[];
+	}
+
+	export interface CommandContext {
+		subcommandName?: string;
+	}
+
+	export interface Command {
+		subcommands: string[];
+		parentName?: string;
+	}
+
+	export interface SapphireClientOptions {
+		subcommands?: SubcommandsPluginOptions;
+	}
+}
+
+declare module 'discord.js' {
+	export interface Client {
+		subcommandsHandler: SubcommandsHandler;
+	}
+}
```

Comment:
In line with having made discord.js an optional peer dependency, this should be relocated to another convenience register. Make sure you augment SapphireClient in this one anyways, though.